core/src/main/java/com/graphhopper/routing/Path.java

```diff
@@ -676,23 +676,30 @@ private boolean isLeavingCurrentStreet(long flag, long prevFlag, int baseNode, i
                 EdgeIterator edgeIter = outEdgeExplorer.setBaseNode(baseNode);
                 while (edgeIter.next()) {
                     if (edgeIter.getAdjNode() != prevNode && edgeIter.getAdjNode() != adjNode) {
-                        String edgename = edgeIter.getName();
-                        long edgeflag = edgeIter.getFlags();
+                        String edgeName = edgeIter.getName();
+                        long edgeFlag = edgeIter.getFlags();
                         // leave the current street || enter a different street
-                        if (isNameSimilar(prevName, edgename) || isNameSimilar(name, edgename)) {
-                            if (checkFlag) {
-                                // leave the current street || enter a different street
-                                if (prevFlag == edgeflag || flag == edgeflag)
-                                    return true;
-                            } else {
-                                return true;
-                            }
+                        if(isTheSameStreet(prevName, prevFlag, edgeName, edgeFlag, checkFlag)
+                                ||isTheSameStreet(name, flag, edgeName, edgeFlag, checkFlag)){
+                            return true;
                         }
                     }
                 }
                 return false;
             }
 
+            private boolean isTheSameStreet(String name1, long flags1, String name2, long flags2, boolean checkFlag){
+                if (isNameSimilar(name1, name2)) {
+                    if (checkFlag) {
+                        if (flags1 == flags2)
+                            return true;
+                    } else {
+                        return true;
```

Comment:
Thanks, yes this make the code a lot better. Now we could even merge the new if with the outer if, but I think this is ok for readability.